# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Rupikaupa2006/Flyrank-ML-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*



I chose the Content Refresh Prioritization lane because it focuses on identifying pages that should be updated to improve their search performance. This interests me because it combines data analysis with machine learning to support better decision-making. As a beginner in machine learning, I believe this lane will help me understand how data-driven models can solve real-world SEO problems while improving my practical ML skills.

In [29]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*



My research question is: Which pages should be prioritized for content refresh to improve their search performance?

This work helps content managers and SEO teams decide which pages to refresh first. If the recommendation is wrong, valuable pages may be ignored while time and resources are spent updating pages that do not need improvement. This can reduce the overall effectiveness of the content refresh strategy.

In [30]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [31]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
!git clone https://github.com/Rupikaupa2006/Flyrank-ML-internship.git

fatal: destination path 'Flyrank-ML-internship' already exists and is not an empty directory.


In [32]:
import os

print(os.listdir("/content/Flyrank-ML-internship/data/raw"))

['content_refresh_anonymized.csv']


In [33]:
import pandas as pd

df = pd.read_csv("/content/Flyrank-ML-internship/data/raw/content_refresh_anonymized.csv")

print(df.shape)
print(df.columns)

(30000, 44)
Index(['content_id', 'client_id', 'search_volume', 'competition',
       'competition_level', 'cpc', 'content_type', 'main_intent', 'word_count',
       'char_count', 'provider_used', 'model_used', 'impressions_90d',
       'clicks_90d', 'pageviews_90d', 'sessions_90d', 'users_90d',
       'engaged_sessions_90d', 'ai_sessions_90d', 'scroll_events_90d',
       'days_with_impressions', 'days_with_sessions', 'impressions_last_30d',
       'clicks_last_30d', 'sessions_last_30d', 'impressions_prev_30d',
       'clicks_prev_30d', 'sessions_prev_30d', 'content_age_days', 'age_tier',
       'age_tier_order', 'days_since_last_update', 'freshness_tier',
       'word_count_tier', 'char_count_tier', 'ctr', 'avg_position',
       'engagement_rate', 'scroll_rate', 'ai_traffic_pct', 'impression_tier',
       'position_tier', 'trend_direction', 'trend_pct'],
      dtype='object')


In [34]:
print("Number of rows:", len(df))
print("Number of columns:", len(df.columns))

print("\nContent Type Distribution:")
print(df["content_type"].value_counts().head())

print("\nTrend Direction Distribution:")
print(df["trend_direction"].value_counts())

Number of rows: 30000
Number of columns: 44

Content Type Distribution:
content_type
keyword article       27207
feedly article         2096
comparison article      697
Name: count, dtype: int64

Trend Direction Distribution:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64




I explored the starter dataset to understand its structure and why this research question is useful. The dataset contains **30,000 rows** and **44 features**. Most pages are **keyword articles (27,207)**, followed by **feedly articles (2,096)** and **comparison articles (697)**. The trend distribution shows that **16,262 pages are declining**, while **5,962 are stable** and **4,388 are improving**. This suggests there are many pages that could potentially benefit from content refresh, making this a meaningful machine learning problem.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [35]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.




This project can identify patterns in historical data and provide decision support for selecting pages that may benefit from content refresh. The results are observational and directional, helping content teams prioritize their work. However, this project cannot prove that refreshing a page will definitely improve its ranking or traffic, and it cannot predict Google's ranking algorithm or establish a causal relationship.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.